### DISCUSSION TITLE
DialogTrigger with Dialog only under certain conditions

### DISCUSSION THREAD

**Discussion by sanbornhilland** [radix-ui/primitives#2468] [Help]
I believe the following case has not been discussed yet. I am curious if it is sensible.

The situation is a button that can exist in two states, `enabled` or `disabled`. When clicking the button to _enable_, the associated action should be immediately performed. The user is not interrupted. However, _disabling_ is potentially destructive. So if the user now clicks the button to _disable_ they should be shown first an `AlertDialog` requiring a confirmation step.

Making this behaviour is straightforward if you control the `open` state of the `AlertDialog` and only open in the case of disabling. However, my question revolves around the use of the `aria-haspopup` attribute that is applied to the `Trigger`. Using a single button element means that it will always have `aria-haspopup` set but one state of the button does not in fact trigger a popup. 

It is unclear to me if `aria-haspopup` _requires_ a popup to be triggered or if it means that a popup _may_ be triggered. The [WAI-ARIA Guidelines](https://w3c.github.io/aria/#aria-haspopup) state:

> [Indicates](https://w3c.github.io/aria/#dfn-indicates) the availability and type of interactive popup [element](https://dom.spec.whatwg.org/#concept-element), such as menu or dialog, that can be triggered by an element.

To me "can be triggered" suggests that a popup _may_ be triggered but not necessarily. Does anyone have any insight here?

The alternative way to do this is to use two buttons. One for the `enable` case and one for the `disable` case. Swap them out when the state changes. But this seems to be like a disaster for focus management, accessibility (knowing what the state of the button is) etc. So it seems to me like the worse option.

So my two questions are

1. Is it okay to *not* trigger a dialog under some conditions with `araia-haspopup`
1. If *no*, is there a reasonable addition we could make to Radix to better support this case?

**Comment by guotie** [comment]
yes, I have same question, but seems nobody care.

**Comment by benoitgrelard** [comment]
I would definitely use 2 buttons for this, one for enabling and another one for disabling (which would be a dialog with its trigger).
